### PR TITLE
hardening/630_update_naming_conventions_sink_docs

### PR DESCRIPTION
**Pull request by frbattid**
* Partially implements issue #630 
* No tests are required
* Assignee @pcoello25

### CODE REVIEW

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md [hunk 1]
```diff
@@ -37,53 +49,84 @@ This is done at the Cygnus Http listeners (in Flume jergon, sources) thanks to [
...
+[Top](#top)
+
+####<a name="section1.2.1"></a>HDFS paths naming conventions
+Since the unique data model accepted for `NGSIHDFSSink` is per entity (see the [Configuration](#section2.1) section for more details), a HDFS folder:
+
+    /user/<hdfs_userame>/<fiware-service>/<fiware-servicePath>/<destination>
+    
+is created (if not existing yet) for each notified entity, where `<hdfs_username>` is a configuration parameter about a HDFS superuser, `<fiware-service>` and `<fiware-servicePath>` are notified as Http headers (or defaulted in NGSIRestHandler), and `<destination>` is the `notified_entities`/`grouped_entities` header value (depending on using or not the grouping rules, see the [Configuration](#section2.1) section for more details) within the Flume event.
+
+Then, fir each notified entity a file named `<destination>.txt` is created (if not yet existing) under the above directory. Again, `<destination>` is the `notified_entities`/`grouped_entities` header value (depending on using or not the grouping rules, see the [Configuration](#section2.1) section for more details) within the Flume event.
```
Comment (pcoello25) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md]: fir -> for

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md [hunk 2]
```diff
@@ -37,53 +49,84 @@ This is done at the Cygnus Http listeners (in Flume jergon, sources) thanks to [
...
+
+Please observe HDFS folders and files follow the [Unix rules](https://en.wikipedia.org/wiki/Filename#Reserved_characters_and_words) about allowed character set. Any forbiden character will be replaced with underscore `_`.
```
Comment (pcoello25) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md]: forbiden -> forbidden

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md [hunk 3]
```diff
@@ -37,53 +49,84 @@ This is done at the Cygnus Http listeners (in Flume jergon, sources) thanks to [
...
+* For each notified attribute, a field named as the attribute is considered. This field will store the attribute values along the time.
+* For each notified attribute, a field named as the concatenation of the attribute name and `_md` is considered. This field will store the attribute's metadata values along the time.
+
+[Top](#top)
+
+####<a name="section1.2.4"></a>CSV row-like storing
+Regarding the specific data stored within the HDFS file, if `file_format` parameter is set to `csv-row` then the notified data is stored attribute by attribute, composing a CSV record for each one of them. Each record contains the following fields:
+
+* `recvTimeTs`: UTC timestamp expressed in miliseconds.
+* `recvTime`: UTC timestamp in human-redable format ([ISO 8601](http://en.wikipedia.org/wiki/ISO_8601)).
```
Comment (pcoello25) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md]: redable -> readable

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md [hunk 4]
```diff
@@ -37,53 +49,84 @@ This is done at the Cygnus Http listeners (in Flume jergon, sources) thanks to [
...
+* `attrType`: Notified attribute type.
+* `attrValue`: In its simplest form, this value is just a string, but since Orion 0.11.0 this can be a JSON object or JSON array.
+* `attrMd`: In this case, the field does not contain the real metadata, but the name of the HDFS file storing such metadata. The reason to do this is the metadata may be an array of any length; each element within the array will be persisted as a single line in the metadata file containing the metadata's name, type and value, all of them separated by the ',' field sepator. There will be a metadata file per each attribute under `/user/<hdfs_userame>/<fiware-service>/<fiware-servicePath>/<destination>_<attrName>_<attrType>/<destination>_<attrName>_<attrType>.txt`
+
+[Top](#top)
+
+####<a name="section1.2.5"></a>CSV column-like storing
+Regarding the specific data stored within the HDFS file, if `file_format` parameter is set to `csv-column` then a single CSV record is composed for the whole notified entity, containing the following fields:
+
+* `recvTime`: UTC timestamp in human-redable format ([ISO 8601](http://en.wikipedia.org/wiki/ISO_8601)).
```
Comment (pcoello25) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md]: redable -> readable :stuck_out_tongue_closed_eyes:

Comment (frbattid) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1047/commits/7dd012b72d08a7853d801433a47602fb3674a327

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md (hunk 3 above)
Comment (frbattid) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1047/commits/7dd012b72d08a7853d801433a47602fb3674a327

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md (hunk 2 above)
Comment (frbattid) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1047/commits/7dd012b72d08a7853d801433a47602fb3674a327

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md (hunk 1 above)
Comment (frbattid) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1047/commits/7dd012b72d08a7853d801433a47602fb3674a327

**Comment by pcoello25** [comment]
LGTM

**Pull request merged**